gapis/api/sync/data.go

```diff
@@ -91,6 +91,11 @@ type Data struct {
 	SyncNodes        []SyncNode
 	CmdSyncNodes     map[api.CmdID]SyncNodeIdx
 	SubcommandLookup *api.SubCmdIdxTrie
+	// UnblockingCommands is a map of what command fully unblocks
+	// some other command. Or to put it another way, the command that answers
+	// the question: If we want to play up to command N, what command do we
+	// have to replay until, in order to prevent a deadlock.
+	UnblockingCommands map[api.CmdID]api.CmdID
```

Comment:
It looks like `UnblockingCommands` handles the case where a submission blocks on a timeline semaphore that is host-signaled later *in the trace*. But what if the trace ended with the command still blocked? I think we need to make sure those commands never get submitted.

Reply:
Chatted offline, but a re-cap here.
This is a correct analysis. However since we only stop at frame boundaries, this should not be a problem in practice. We have a lot of other problems to fix if we ever end up in a case where an EOF is not a safe place to block the GPU.